server/neptune/gateway/event/check_run_handler.go

```diff
@@ -102,27 +121,70 @@ func (h *CheckRunHandler) signalPlanReviewWorkflowChannel(ctx context.Context, e
 	return nil
 }
 
-func (h *CheckRunHandler) signalUnlockWorkflowChannel(ctx context.Context, event CheckRun) error {
-	// Parse out root from check run name
-	matches := checkRunRegex.FindStringSubmatch(event.Name)
-	if len(matches) != 2 {
-		return fmt.Errorf("unable to determine root name")
-	}
-	rootName := matches[checkRunRegex.SubexpIndex("name")]
-	ctx = context.WithValue(ctx, contextInternal.ProjectKey, rootName)
-	err := h.TemporalClient.SignalWorkflow(
+func (h *CheckRunHandler) signalUnlockWorkflowChannel(ctx context.Context, event CheckRun, rootName string) error {
+	err := h.DeploySignaler.SignalWorkflow(
 		ctx,
 		// deploy workflow id is repo||root (the name of the check run is the root)
 		buildDeployWorkflowID(event.Repo.FullName, rootName),
 		// keeping this empty is fine since temporal will find the currently running workflow
 		"",
 		workflows.DeployUnlockSignalName,
 		workflows.DeployUnlockSignalRequest{
-			User: event.User,
+			User: event.User.Username,
 		})
 	if err != nil {
 		return errors.Wrapf(err, "signaling workflow with id: %s", event.ExternalID)
 	}
 	h.Logger.InfoContext(ctx, fmt.Sprintf("Signaled workflow with id %s to unlock", event.ExternalID))
 	return nil
 }
+
+func (h *CheckRunHandler) handleReRequestedRun(ctx context.Context, event CheckRun, rootName string) error {
+	// Block force applies
+	if event.Branch != event.Repo.DefaultBranch {
+		h.Logger.DebugContext(ctx, "dropping event branch unexpected ref")
+		return nil
+	}
+	builderOptions := BuilderOptions{
+		RepoFetcherOptions: github.RepoFetcherOptions{
+			ShallowClone: true,
+		},
+		FileFetcherOptions: github.FileFetcherOptions{
+			Sha: event.HeadSha,
+		},
+	}
+	rootCfgs, err := h.RootConfigBuilder.Build(ctx, event.Repo, event.Branch, event.HeadSha, event.InstallationToken, builderOptions)
```

Comment:
Worth noting that if we re-run a revision that has x roots changed, we're going to get x events and therefore clone x times yeah?  It's probably fine but just making sure we're aware of what could potentially happen.

Reply:
Yes, unfortunately check run looks to send separate events for each "failed" health check.